### DISCUSSION TITLE
Are both XML and DB storage considered persistent storage?

### DISCUSSION THREAD

**Discussion by Pilloo** [quartznet/quartznet#2615] [Q&A]
Hi! I'm trying to get a grasp on Quartz, more specifically with persistent storage and file-based configuration, but I got confused with the concepts. From what I've read, my understanding is that XML files are primarily used to define new jobs and DBs are used as a "backup" to keep track of jobs, triggers and calendars. So, e.g., I could write a new entry in the XML file and the scheduler would update the database automatically. Is that correct? If no, how does it work? I really appreciate your advice. Thanks!

**Comment by lahma** [comment]
XML configuration can be considered a trigger/job definition syncing source. Definitions from XML are synced to scheduler (and there can be a file watcher to dynamically update based on changes). DB persistence is needed if you need fault tolerance, clustering or job state to be persisted - XML doesn't handle these concerns.
